bundles/replication/src/main/java/com/adobe/cq/commerce/demandware/replication/transport/ContentAssetFolderPlugin.java

```diff
@@ -76,8 +80,10 @@ public boolean deliver(JSONObject delivery, AgentConfig config, ReplicationLog l
                         log.info("Assign content asset %s to folder %s (%s)", id, folder, action.getType().getName());
 
                         // construct URL for folder assignment
-                        final String transportUriBuilder = DemandwareClient.DEFAULT_SCHEMA + demandwareClient.getEndpoint() + getOCApiPath() + getOCApiVersion() + constructEndpointURL(
-                                delivery.getString(DemandwareCommerceConstants.ATTR_API_ENDPOINT), folder, delivery);
+                        final String endpoint = clientProvider.getClientForSpecificInstance(instanceId.getInstanceId(config))
+                                .map(DemandwareClient::getEndpoint).orElse(StringUtils.EMPTY);
```

Comment:
we need an endpoint. this is not optional. it must not be empty.

Reply:
We can throw ReplicationException if Demandware Client not found. I didn't request this before because it was not thrown in the initial implementation and we tried to change as little as possible.

Additionally, getting the Demandware Client can be moved out of the for loop.